source/src/microapp/cs_MicroappController.cpp

```diff
@@ -364,31 +371,25 @@ void MicroappController::softInterruptGpio(uint8_t pin) {
 	cmd->header.id           = digitalPinToInterrupt(pin);
 	cmd->pin                 = digitalPinToInterrupt(pin);
 	// Resume microapp so it can pick up this command.
-	LOGi("GPIO interrupt on virtual pin %i", cmd->pin);
+	LOGv("Incoming GPIO interrupt for microapp on virtual pin %i", cmd->pin);
 	softInterrupt();
 }
 
 /*
- * Communicate mesh message towards microapp.
+ * Communicate BLE message towards microapp.
  */
 void MicroappController::softInterruptBle(scanned_device_t* bluenetBleDevice) {
 
-#ifdef DEVELOPER_OPTION_THROTTLE_BY_RSSI
+	// Throttle by rssi by default to limit number of scanned devices forwarded
```

Comment:
Fine by me. Can I ask why?

Reply:
Sure, I see the throttling as a hack. Very necessary for now, but this is not how it should be solved in the end. 

First. Currently, there's a bottleneck between bluenet and the microapp(s) by populating a single buffer and then switching context. If we could reserve a large ram section for this, we might populate a much larger buffer with messages and then switch context for example. Especially for the nRF52840 we can start playing with a larger inter-process space.

Second. If almost nothing happens in the soft interrupt at the microapp side, it should be possible to go back and forth thousand times per second. This only starts to go haywire when people misuse interrupts. For example, we support `Serial.write` within the interrupt handler. Of course, a mature embedded developer would never do such a thing. Rather than throttling by RSSI, which is quite a random thing to do, we should educate the microapp developer to write their code better. For example, somehow warn for each call towards bluenet within an interrupt handler except for a few.

Reply:
Was this comment meant for the conversation below? I will reply there